### PR TITLE
Fix repo home UI when there is no repo description

### LINKED ISSUE #30502: Non-admin user doesn't show repo dummy description
(just to record, haven't looked into it, the UI looks pretty strange because there is a search input)

image

image

### PR DESCRIPTION
**Pull request by wxiaoguang**
Fix #30502 by a new approach.

image

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: templates/repo/home.tmpl [hunk 1]
```diff
@@ -5,18 +5,14 @@
...
-					<input name="q" value="{{.Keyword}}" placeholder="{{ctx.Locale.Tr "search.code_kind"}}">
-					{{template "shared/search/button"}}
-				</div>
-			</form>
+		<div class="repo-description gt-word-break">
+			{{- $description := .Repository.DescriptionHTML ctx -}}
+			{{- if $description -}}
+				{{- $description | RenderCodeBlock -}}
+			{{- else if .IsRepositoryAdmin -}}
+				<span class="no-description text italic">{{ctx.Locale.Tr "repo.no_desc"}}</span>
```
Comment (silverwind) [inline templates/repo/home.tmpl]:
```suggestion
				<span class="no-description">{{ctx.Locale.Tr "repo.no_desc"}}</span>
```

Imho, I prefer no italic here and the previous class was broken a long time.

Same change is also in https://github.com/go-gitea/gitea/pull/30576 coincidentially.

Comment (wxiaoguang) [reply templates/repo/home.tmpl]: It is the same as:

image

**Comment by wxiaoguang** [comment]
> What is the point of the `.IsRepositoryAdmin` condition here?

Don't show "no description" to non-admin users, I think it's not bad.

File: templates/repo/home.tmpl (hunk 1 above)
Comment (wxiaoguang) [reply templates/repo/home.tmpl]:
```suggestion
				<span class="no-description tw-italic">{{ctx.Locale.Tr "repo.no_desc"}}</span>
```

**Comment by silverwind** [comment]
> > What is the point of the `.IsRepositoryAdmin` condition here?
> 
> Don't show "no description" to non-admin users, the same as before, I think it's not bad.

But why? I don't understand the reason why the rendering of a placeholder would depend on user permission level. It makes no sense.

**Comment by wxiaoguang** [comment]
> > > What is the point of the `.IsRepositoryAdmin` condition here?
> > 
> > 
> > Don't show "no description" to non-admin users, the same as before, I think it's not bad.
> 
> But why? I don't understand the reason why the rendering of a placeholder would depend on user permission level. It makes no sense.

I think the "no description" here makes no sense to end users, and it makes the UI looks ugly. So it's not bad to hide it for non-admin users.

It is different from the "no description"  in an empty issue, without that dummy message, the empty issue UI looks strange, so it is good to have it for everyone.

**Comment by silverwind** [comment]
I do agree we should hide `No description` for all users like GitHub also does, if we can make the edit UI still available to users with that permission.

**Comment by wxiaoguang** [comment]
Removed the "no-description" in 827565337990ead469a1cdfc7fb2c912a6b7d851

**Review by silverwind — APPROVED** [review]

**Comment by yp05327** [comment]
image
The search code box seems strange ( no space to the filename, maybe it should be in the right side? also may need considering long file path) in file view page, maybe related to this PR?

**Comment by wxiaoguang** [comment]
-> Fix code search input for different views #30678

**Pull request merged**